File: go/docmost/services/health_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
)

// HealthAPI implements Health API.
type HealthAPI struct {
	http *httpclient.HTTPClient
}

func NewHealthAPI(http *httpclient.HTTPClient) *HealthAPI {
	return &HealthAPI{http: http}
}

func (a *HealthAPI) HealthCheck(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Get(ctx, "health", nil)
}

func (a *HealthAPI) Liveness(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Get(ctx, "health/live", nil)
}

File: python/docmost/services/health_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse


class HealthApi:
    """Health API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def health_check(self) -> ApiResponse[Any]:
        return self._http.get("health")

    def liveness(self) -> ApiResponse[Any]:
        return self._http.get("health/live")

